### PR TITLE
Add documentation for the Application Insights sample

### PR DESCRIPTION
**Pull request by paselem**

### CODE REVIEW

File: CSharp/ArticleProjects/ApplicationInsights/README.md [hunk 1]
```diff
@@ -0,0 +1,392 @@
+---
+title: Monitoring and debugging Azure Batch applications with Application Insights - Azure | Microsoft Docs
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/README.md]: Does this stuff show up nicely in a rendered markdown on GitHub?

Comment (matthchr) [reply CSharp/ArticleProjects/ApplicationInsights/README.md]: It seems like that's a no, it doesn't -- can we make it look pretty?

File: CSharp/ArticleProjects/ApplicationInsights/README.md [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+with custom metrics and tracing. Integrating Application Insights with your 
+Azure Batch application allows developers to gain deep insights into behaviors 
+and investigate issues in near-real-time.
+
+This article shows how to add and configure the Application Insights library 
+into your solution and instrument your application code. Futhermore it provides
+examples on how to monitor your application via the Azure portal and build 
+custom dashboards.
+
+*Should this be added in the learn more section?*
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/README.md]: It seems that this was authored for acom and doesn't render as beautifully here?

i.e. these comments show up

Same below where `!WARNING` shows up

File: CSharp/ArticleProjects/ApplicationInsights/README.md [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+
+We also inherently track exceptions. This sample purposely leaves out exception 
+handling to see how Application Insights automatically reports unhandled 
+exceptions for us and significantly improves the debugging experience. The 
+following sample illustrates how to use these methods.
+
+```csharp
+public void CountWords(string blobName, int numTopN, string storageAccountName, string storageAccountKey)
+{
+    // simulate exception for some randome set of tasks
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/README.md]: minor: Randome -> Random

File: CSharp/ArticleProjects/ApplicationInsights/README.md [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+FileToStage applicationInsightsConfig = new FileToStage(AIConfig, stagingStorageAccount);
+FileToStage applicationInsightsDll = new FileToStage(AIDllName, stagingStorageAccount);
+FileToStage applicationInsightsCollectorDll = new FileToStage(AIDependencyCollectorName, stagingStorageAccount);
+FileToStage applicationInsightsInterceptorDll = new FileToStage(AIInterceptDllAgentName, stagingStorageAccount);
+FileToStage applicationInsightsPerfCounterCollectorDll = new FileToStage(AIPerfCounterCollectorName, stagingStorageAccount);
+FileToStage applicationInsightsServerTelemetryDll = new FileToStage(AIServerTelemetryName, stagingStorageAccount);
+FileToStage applicationInsightsWindowsServerDll = new FileToStage(AIWindowsServerName, stagingStorageAccount);
+```
+
+> Th FileToStage method is a helper function in the code
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/README.md]: The

File: CSharp/ArticleProjects/ApplicationInsights/README.md [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+data flowing and getting logged. In this article we'll only touch on a few 
+features, but feel free to explore the full feature set provided by the 
+Application Insights service.
+
+### View live stream data
+
+The following screenshot shows how we can view live data coming from the 
+Compute Nodes in the pool, for example the CPU usage per Compute Node.
+
+![Live stream compute node data](./media/batch-monitoring-with-application-insights/ApplicationInsightsLiveStream.png)
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/README.md]: I think things like `!` prefix don't render here well

File: CSharp/ArticleProjects/ApplicationInsights/README.md (hunk 1 above)
Comment (paselem) [reply CSharp/ArticleProjects/ApplicationInsights/README.md]: Removed table

File: CSharp/ArticleProjects/ApplicationInsights/README.md (hunk 2 above)
Comment (paselem) [reply CSharp/ArticleProjects/ApplicationInsights/README.md]: Seems to render correctly.

File: CSharp/ArticleProjects/ApplicationInsights/README.md (hunk 4 above)
Comment (paselem) [reply CSharp/ArticleProjects/ApplicationInsights/README.md]: Fixed

File: CSharp/ArticleProjects/ApplicationInsights/README.md (hunk 2 above)
Comment (matthchr) [reply CSharp/ArticleProjects/ApplicationInsights/README.md]: This is a TODO though right? Should we remove it?

**Pull request merged**